src/course-outline/data/thunk.js

```diff
@@ -24,3 +33,58 @@ export function fetchCourseOutlineIndexQuery(courseId) {
     }
   };
 }
+
+export function fetchCourseLaunchQuery({
+  courseId,
+  gradedOnly = true,
+  validateOras = true,
+  all = true,
+}) {
+  return async (dispatch) => {
+    try {
+      const data = await getCourseLaunch({
+        courseId, gradedOnly, validateOras, all,
+      });
+      dispatch(fetchStatusBarSelPacedSuccess({ isSelfPaced: data.isSelfPaced }));
+      dispatch(fetchStatusBarChecklistSuccess(getCourseLaunchChecklist(data)));
+
+      return true;
+    } catch (error) {
+      return false;
+    }
+  };
+}
+
+export function fetchCourseBestPracticesQuery({
+  courseId,
+  excludeGraded = true,
+  all = true,
+}) {
+  return async (dispatch) => {
+    try {
+      const data = await getCourseBestPractices({ courseId, excludeGraded, all });
+      dispatch(fetchStatusBarChecklistSuccess(getCourseBestPracticesChecklist(data)));
+
+      return true;
+    } catch (error) {
+      return false;
+    }
+  };
+}
+
+export function enableCourseHighlightsEmailsQuery(courseId) {
+  return async (dispatch) => {
+    dispatch(updateSavingStatus({ status: RequestStatus.IN_PROGRESS }));
```

Comment:
let's use `dispatch(updateSavingStatus({ status: RequestStatus.PENDING }));`
for `saving` before starting `POST, DELETE...`
for `loading` `RequestStatus.IN_PROGRESS`

Reply:
Done